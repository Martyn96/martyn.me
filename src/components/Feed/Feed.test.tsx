import React from 'react';
import renderer from 'react-test-renderer';
import { Feed } from '.';

describe('feed', () => {
  const props = {
    edges: [
      {
        node: {
          fields: {
            slug: '/test_0',
          },
          frontmatter: {
            date: '2016-09-01',
            description: 'test_0',
            title: 'test_0',
          },
          timeToRead: 3,
          id: 'test-123',
          html: '<p>test</p>',
        },
      },
      {
        node: {
          fields: {
            slug: '/test_1',
          },
          frontmatter: {
            date: '2016-09-01',
            description: 'test_1',
            title: 'test_1',
          },
          timeToRead: 1,
          id: 'test-321',
          html: '<p>test</p>',
        },
      },
    ],
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Feed {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
