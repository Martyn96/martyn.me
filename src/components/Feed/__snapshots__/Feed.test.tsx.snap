// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div
  className="feed"
>
  <a
    className="feed__item"
    href="/test_0"
  >
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime="September 1, 2016"
      >
        September 1, 2016
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-readtime"
      >
        3
         minute
        s
         read
      </span>
    </div>
    <h2
      className="feed__item-title"
    >
      test_0
    </h2>
    <p
      className="feed__item-description"
    >
      test_0
    </p>
  </a>
  <a
    className="feed__item"
    href="/test_1"
  >
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime="September 1, 2016"
      >
        September 1, 2016
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-readtime"
      >
        1
         minute
        
         read
      </span>
    </div>
    <h2
      className="feed__item-title"
    >
      test_1
    </h2>
    <p
      className="feed__item-description"
    >
      test_1
    </p>
  </a>
</div>
`;
